byway/src/main/webapp/home.js

```diff
@@ -86,22 +87,46 @@ function showIncompleteTrip(tripNum, trip, isDestinationsMissing, isInterestsMis
  * @param {Number} tripNum 
  * @param {Trip} trip
  */
-function showCompleteTrip(tripNum, trip){
+async function showCompleteTrip(trip){
   let container = document.getElementById("past-trips-container");
   let pastTrip = document.createElement('div');
   pastTrip.className = "past-trip";
   let title = document.createElement('a');
+  title.id = "title-" + trip.keyString;
   pastTrip.append(title);
   let mapContainer = document.createElement('div');
   mapContainer.className = 'map';
-  mapContainer.id = "map-" + trip.keyString
+  mapContainer.id = "map-" + trip.keyString;
   pastTrip.append(mapContainer);
   container.append(pastTrip);
   initMap(trip.start, trip.start, trip.route, trip.keyString);
-  title.innerText = "Trip #" + tripNum;
+  let titleString = await constructTripTitle(trip);
+  title.innerText = titleString;
   title.href = configureTripKeyForPath(trip.keyString, "/routepage.html");
 }
 
+async function constructTripTitle(trip){
+  //let title = document.getElementById('title-' + keyString);
+  //title.innerText = "";
+  for(let i = 0; i<30; i++){
+      try{
+        const promises = trip.destinations.map(async (destination) => {
```

Comment:
Given the fact that you have to query the Maps API to get the name of places, It may be easier to store this as some string on the Trip object instead. It is not difficulty to add a new field to the Java object or datastore object. This way you wouldn't need this convoluted logic to ascertain the start and end names. 

The value of this string Title could be generated on the first step when you select the start and end points while you already have the string names. Or you could let the user enter one manually. Or auto generate a title based on the date, etc.

Reply:
I updated the function to simply loop through and find the placeId of each destination and add to the title, so the logic is not too confusing anymore. I did think about creating a new field, but since alot of pages rely on creating trips, I did not want to mess up anything by adding a new field to the constructor. Also, since we are running out of time, I figured this would be the quickest way to get some sort of title, even though it would be cool to have the user enter their title manually